addons/portal/models/mail_message.py

```diff
@@ -16,8 +16,11 @@ def portal_message_format(self):
 
     def _portal_message_format(self, fields_list):
         vals_list = self._message_format(fields_list)
+        note_id = self.env['ir.model.data'].xmlid_to_res_id('mail.mt_note')
         IrAttachmentSudo = self.env['ir.attachment'].sudo()
         for vals in vals_list:
+            message_sudo = self.browse(vals['id']).sudo().with_prefetch(self.ids)
+            vals['is_note'] = message_sudo.subtype_id.id == note_id
```

Comment:
Is the subtype not available within vals?
Seems harsh to browse within a loop, even if we use the "with_prefetch" trick.

(I see you copy/pasted from other overrides of "message_format", but it seems like "subtype_id" is included in the vals if I look at the base docstring of message_format).